From: openKylin Developers <packaging@lists.openkylin.top>
Date: Tue, 2 Jun 2026 18:36:01 +0800
Subject: fix FTBFS caused by C23 const and glibc 2.43

Origin: upstream, https://github.com/nss-dev/nss/commit/5c4ef88cc30ec1a171bc207dfb8e03c8dd6471bc
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/nss/+bug/2142897

Fixes:
    In file included from lowhash_vector.c:65:
    genload.c: In function 'loader_LoadLibInReferenceDir':
    genload.c:92:7: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
       92 |     c = strrchr(referencePath, PR_GetDirectorySeparator());
          |       ^

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>

Differential Revision: https://phabricator.services.mozilla.com/D281507
---
 nss/lib/freebl/genload.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nss/lib/freebl/genload.c b/nss/lib/freebl/genload.c
index 832deb5..e3231e3 100644
--- a/nss/lib/freebl/genload.c
+++ b/nss/lib/freebl/genload.c
@@ -85,7 +85,7 @@ loader_LoadLibInReferenceDir(const char* referencePath, const char* name)
 {
     PRLibrary* dlh = NULL;
     char* fullName = NULL;
-    char* c;
+    const char* c;
     PRLibSpec libSpec;
 
     /* Remove the trailing filename from referencePath and add the new one */
